src/main/java/org/jabref/gui/linkedfile/LinkedFileEditDialogViewModel.java

```diff
@@ -86,12 +94,29 @@ public void openBrowseDialog() {
                 .withInitialFileName(fileName)
                 .build();
 
-        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(path -> {
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(fileToAdd -> {
             // Store the directory for next time:
-            filePreferences.setWorkingDirectory(path);
-            link.set(relativize(path));
 
-            setExternalFileTypeByExtension(link.getValueSafe());
+            if (FileUtil.detectBadFileName(fileToAdd.toString())) {
+                String newFilename = FileNameCleaner.cleanFileName(fileToAdd.getFileName().toString());
+
+                boolean correctButtonPressed = dialogService.showConfirmationDialogAndWait(Localization.lang("File \"%0\" cannot be added!", fileToAdd.getFileName()),
+                        Localization.lang("Illegal characters in the file name detected.\nFile will be renamed to \"%0\" and added.", newFilename),
+                        Localization.lang("Rename and add"));
+
+                if (correctButtonPressed) {
+                    Path correctPath = fileToAdd.resolveSibling(newFilename);
+                    try {
+                        Files.move(fileToAdd, correctPath);
+                        link.set(relativize(correctPath));
+                        filePreferences.setWorkingDirectory(correctPath);
+                        setExternalFileTypeByExtension(link.getValueSafe());
```

Comment:
Can we unit test that this part is executed when it should? We can mock  dialogService.showConfirmationDialogAndWait I think.

Reply:
The problem is that this is executed inside the showFileOpenDialog